test/jdk/com/sun/jdi/TestClassUnloadEvents.java

```diff
@@ -117,30 +124,62 @@ private static boolean classUnloadingOccurred(Process p) throws IOException {
             classUnloadRequest.addClassFilter(CLASS_NAME_PREFIX + "*");
             classUnloadRequest.enable();
 
+            ClassUnloadRequest classUnloadRequest_alt = vm.eventRequestManager().createClassUnloadRequest();
+            classUnloadRequest_alt.addClassFilter(CLASS_NAME_ALT_PREFIX + "*");
+            classUnloadRequest_alt.enable();
+
             EventSet eventSet = null;
             boolean exited = false;
             while (!exited && (eventSet = vm.eventQueue().remove()) != null) {
+                System.out.println("EventSet: " + eventSet);
                 for (Event event : eventSet) {
                     if (event instanceof ClassUnloadEvent) {
                         String className = ((ClassUnloadEvent)event).className();
-                        unloadedSampleClasses.add(className);
+
+                        // The unloaded class should always match CLASS_NAME_PREFIX.
+                        if (className.indexOf(CLASS_NAME_PREFIX) == -1) {
+                            throw new RuntimeException("FAILED: Unexpected unloaded class: " + className);
+                        }
+
+                        // Unloaded classes with ALT names should only occur on the classUnloadRequest_alt.
+                        if (event.request() == classUnloadRequest_alt) {
+                            unloadedSampleClasses_alt.add(className);
+                            if (className.indexOf(CLASS_NAME_ALT_PREFIX) == -1) {
+                                throw new RuntimeException("FAILED: non-alt class unload event for classUnloadRequest_alt.");
+                            }
+                        } else {
+                            unloadedSampleClasses.add(className);
+                        }
+
+                        // If the unloaded class matches the ALT prefix, then we should have
+                        // unload events in this EventSet for each of the two ClassUnloadRequesta.
+                        int expectedEventSetSize;
+                        if (className.indexOf(CLASS_NAME_ALT_PREFIX) != -1) {
+                            expectedEventSetSize = 2;
+                        } else {
+                            expectedEventSetSize = 1;
+                        }
+                        if (eventSet.size() != expectedEventSetSize) {
+                            throw new RuntimeException("FAILED: Unexpected eventSet size: " + eventSet.size());
+                        }
                     }
 
                     if (event instanceof VMDeathEvent) {
                         exited = true;
                         break;
                     }
                 }
-                vm.resume();
+                eventSet.resume();
             }
-        } catch (Exception e) {
+        } catch (InterruptedException | VMCannotBeModifiedException | IOException | IllegalConnectorArgumentsException | VMStartException e) {
```

Comment:
I think you should remove the try/catch and just declare that this method throws Exception. That seems to be what other tests are doing.

Reply:
Done.